xarray/core/variable.py

```diff
@@ -2710,7 +2723,7 @@ def values(self, values):
             f"Please use DataArray.assign_coords, Dataset.assign_coords or Dataset.assign as appropriate."
         )
 
-    def chunk(self, chunks={}, name=None, lock=False):
+    def chunk(self, chunks={}, name=None, lock=False, inline_array=False):
```

Comment:
What is the point of this function if it doesn't do anything?

Reply:
It means that `Dataset.chunk` doesn't have to specifically deal with IndexVariable (convenient!) but is the cause of #6204